File: 905-sort-array-by-parity/905-sort-array-by-parity.java
class Solution {
    public int[] sortArrayByParity(int[] nums) {
        
        int i=0;
        int j=nums.length-1;
        while(i<=j)
        {
            if(nums[i]%2!=0)
            {
                if(nums[j]%2==0)
                {
                    int temp=nums[i];
                    nums[i]=nums[j];
                    nums[j]=temp;
                    i++;
                    j--;
                }
                else
                    j--;
            }
            else
                i++;
        }
        return nums;
    }
}

File: 905-sort-array-by-parity/905-sort-array-by-parity.cpp
class Solution {
public:
    vector<int> sortArrayByParity(vector<int>& A) {
        int size=A.size();
        
        int read=0;
        int write=0;
        for(read=0;read<size;read++)
        {
            if(A[read]%2==0)
            {
                int temp=A[write];
                A[write++]=A[read];
                A[read]=temp;
            }
        }
         
        return A;
    }
};